### PR TITLE
RestController for requesting releases

### PR DESCRIPTION
**Pull request by hedgehog1833**
Rest controller for Releases implemented

### CODE REVIEW

**Comment by hedgehog1833** [comment]
[Create Releases Rest Controller](https://trello.com/c/iG1c7JMD/78-create-releases-rest-controller)

File: src/main/java/com/metalr2/service/releases/ReleasesServiceImpl.java [hunk 1]
```diff
@@ -32,19 +32,19 @@ public ReleasesServiceImpl(RestTemplate restTemplate, @Value("${metal.release.bu
...
-    ResponseEntity<ReleasesResponse> responseEntity = restTemplate.postForEntity(allReleasesUrl, requestEntity, ReleasesResponse.class);
+  public List<ReleaseDto> getReleases(ReleasesButlerRequest request) {
+    HttpEntity<ReleasesButlerRequest> requestEntity = createHttpEntity(request);
+    ResponseEntity<ReleasesButlerResponse> responseEntity = restTemplate.postForEntity(allReleasesUrl, requestEntity, ReleasesButlerResponse.class);
 
-    ReleasesResponse response = responseEntity.getBody();
+    ReleasesButlerResponse response = responseEntity.getBody();
     if (response == null || responseEntity.getStatusCode() != HttpStatus.OK || response.getReleases().isEmpty()) {
       return Collections.emptyList();
     }
```
Comment (DanielW1987) [inline src/main/java/com/metalr2/service/releases/ReleasesServiceImpl.java]: This is a _should not happen_, right? Some logging with WARN level would be nice.

File: src/main/java/com/metalr2/web/dto/releases/ReleasesButlerRequest.java [hunk 2]
```diff
@@ -13,12 +14,13 @@
...
 @AllArgsConstructor
 @ToString
+@EqualsAndHashCode
 @JsonPropertyOrder({
     "dateFrom",
     "dateTo",
     "artists"
 })
-public class ReleasesRequest {
+public class ReleasesButlerRequest {
```
Comment (DanielW1987) [inline src/main/java/com/metalr2/web/dto/releases/ReleasesButlerRequest.java]: I think `ButlerReleaseRequest` fits better.

File: src/main/java/com/metalr2/web/dto/releases/ReleasesButlerResponse.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
 @EqualsAndHashCode
 @ToString
 @NoArgsConstructor
-public class ReleasesResponse {
+public class ReleasesButlerResponse {
```
Comment (DanielW1987) [inline src/main/java/com/metalr2/web/dto/releases/ReleasesButlerResponse.java]: I think `ButlerReleaseResponse` fits better.

File: src/main/java/com/metalr2/web/dto/request/ReleasesRequest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import javax.validation.constraints.NotNull;
+import java.time.LocalDate;
+import java.util.List;
+
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
+@EqualsAndHashCode
+public class ReleasesRequest {
```
Comment (DanielW1987) [inline src/main/java/com/metalr2/web/dto/request/ReleasesRequest.java]: I think `DetectorReleaseRequest` fits better.

File: src/main/java/com/metalr2/web/dto/response/ReleasesResponse.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+@NoArgsConstructor
+public class ReleasesResponse {
```
Comment (DanielW1987) [inline src/main/java/com/metalr2/web/dto/response/ReleasesResponse.java]: I think `DetectorReleaseResponse` fits better.

File: src/main/java/com/metalr2/service/releases/ReleasesServiceImpl.java (hunk 1 above)
Comment (hedgehog1833) [reply src/main/java/com/metalr2/service/releases/ReleasesServiceImpl.java]: You think so? Having no results to a requested input doesn't sound like a should not happen. We could split this and only warn if the status is not OK, but that would be kind of ugly^^

Comment (DanielW1987) [reply src/main/java/com/metalr2/service/releases/ReleasesServiceImpl.java]: `response == null` is a should not happen.
Any status but OK should be logged.
Empty releases list is something that can happen and it's OK.

**Pull request merged**